flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -440,7 +453,11 @@ public void dropTempCatalogFunction(ObjectIdentifier identifier, boolean ignoreI
 
         CatalogFunction fd = tempCatalogFunctions.remove(normalizedName);
 
-        if (fd == null && !ignoreIfNotExist) {
+        if (fd != null) {
```

Comment:
What's the difference between `dropTempCatalogFunction` and `dropTemporaryCatalogFunction`? Should we unify them?

Reply:
Yeah I'll unify them.